### PR TITLE
utils and pathname moved

### PR DESCRIPTION
**Pull request by AnastasiaSulyagina**
### Changes to the core
- [x] Followed [hacking.md](https://github.com/caskroom/homebrew-cask/blob/master/doc/development/hacking.md).

---

Moved utils from hb-fork, deleted pathname, removed duplicated code from hbc/utils. 
_System methods and curl will be deleted in next PR_

cc @MikeMcQuaid

### CODE REVIEW

**Comment by AnastasiaSulyagina** [comment]
Worked on my machine, will see what can be wrong

**Comment by MikeMcQuaid** [comment]
This issue can be worked around but the main problem is the CI is running on Linux but pretending to  be a Mac. This isn't something Homebrew supports and may break at some point in future when our generic OS/Linux support improves.

https://github.com/caskroom/homebrew-cask/blob/master/.travis.yml#L13-L17
For now adding `HOMEBREW_OSX_VERSION=$OSX` to the `env` lines may be enough.

File: lib/hbc/utils.rb [hunk 1]
```diff
@@ -67,6 +83,22 @@ def puts_columns(items, star_items = [])
   puts Hbc::Utils.stringify_columns(items, star_items)
 end
 
+def curl(*args)
```
Comment (MikeMcQuaid) [inline lib/hbc/utils.rb]: This probably also wants a `TODO: delete`

File: lib/hbc/utils.rb [hunk 2]
```diff
@@ -33,22 +33,38 @@ def tty?
 end
 
 # global methods
-
-# originally from Homebrew
-def ohai(title, *sput)
-  title = Tty.truncate(title) if $stdout.tty? && !Hbc.verbose
-  puts "#{Tty.blue.bold}==>#{Tty.reset.bold} #{title}#{Tty.reset}"
-  puts sput unless sput.empty?
+# TODO: delete
```
Comment (MikeMcQuaid) [inline lib/hbc/utils.rb]: Might also want to tweak this to be `TODO: replace with Homebrew's utils version when fixed issue handling non-string args`

Comment (AnastasiaSulyagina) [inline lib/hbc/utils.rb]: It's ok with non-string args already as far as I know. But I'll change it, ok.

Comment (MikeMcQuaid) [inline lib/hbc/utils.rb]: Cool, sorry, that was just an example: whatever the reason is 😉

**Comment by AnastasiaSulyagina** [comment]
> /Users/travis/build.sh: line 109: shell_session_update: command not found

It would be great if somebody could explain why it failed only on 10.11 after it finished with 0.

**Comment by jawshooah** [comment]
See rvm/rvm#3627 and my attempt to address it in #22774.

File: lib/hbc/utils.rb [hunk 3]
```diff
@@ -83,6 +83,7 @@ def puts_columns(items, star_items = [])
   puts Hbc::Utils.stringify_columns(items, star_items)
 end
 
+# TODO: delete
```
Comment (MikeMcQuaid) [inline lib/hbc/utils.rb]: When to delete this one 😀

Comment (AnastasiaSulyagina) [inline lib/hbc/utils.rb]: When I figure out how to reference correct curl_args. Decided that such message is not clear enough to put in comments

Comment (MikeMcQuaid) [inline lib/hbc/utils.rb]: Definitely worth adding something in even if unclear. Unclear trumps empty 😉

**Comment by AnastasiaSulyagina** [comment]
Saw that #22774 is merged, updated to see if checks pass

**Comment by jawshooah** [comment]
Thanks!

**Comment by toonetown** [comment]
For some reason, this change makes all cask `ohai` commands displayed as white for me - instead of bold black.  Reverting this change fixes the problem.

The `ohai` commands issued by homebrew formulas are still the correct color.  Just the cask ones.  You can see this in action by running `brew cask info zulu` (or any other cask that has "Caveats") and notice the incorrect color of the `Caveats` line.

**Comment by jawshooah** [comment]
@toonetown Please create a separate issue. I believe I know the cause, but I'd prefer not to pollute this thread with (mostly) unrelated discussion.

**Pull request merged**